cloud/storage/core/libs/aio/service.cpp

```diff
@@ -194,19 +194,16 @@ class TAIOService final
         const TVector<TArrayRef<char>>& buffers,
         TFileIOCompletion* completion) override
     {
-        auto req = std::make_unique<iocb>();
+        static_assert(sizeof(TArrayRef<char>) == sizeof(iovec));
 
-        TVector<iovec> iov(buffers.size());
-        for (ui32 i = 0; i < buffers.size(); ++i) {
-            iov[i].iov_base = static_cast<void*>(buffers[i].data());
-            iov[i].iov_len = buffers[i].size();
-        }
+        auto req = std::make_unique<iocb>();
 
         io_prep_preadv(
             req.get(),
             file,
-            iov.data(),
-            iov.size(),
+            // TArrayRef has the same layout as iovec
+            std::bit_cast<iovec*>(buffers.data()),
```

Comment:
looks very risky the layout of ArrayRef can change elsewhere the code will compile and we will pass invalid data.
Also I am not sure how much performance we save by eliminate copying the  iovec for small io's the iovec size is small for large io's it could be 1k size for 1M requests. But the overhead of creating iovec is very negligible relative to 1M buffer I/O